File: legacy/WebApp/src/api/booking.ts
// WebApp/src/api/booking.ts
import apiClient from './apiClient';

interface CreateBookingDto {
  serviceId: number;
  availabilitySlotId: number;
}

interface BookingDetailsDto {
  id: number;
  customerId: number;
  customerName: string;
  providerId: number;
  providerName: string;
  serviceId: number;
  serviceName: string;
  slotStartTime: string;
  slotEndTime: string;
  status: string; // Assuming enum or string representation
}

export const createBooking = async (bookingData: CreateBookingDto): Promise<BookingDetailsDto> => {
  const response = await apiClient.post<BookingDetailsDto>('/api/bookings', bookingData);
  return response.data;
};

export const fetchMyBookings = async (customerId: number): Promise<BookingDetailsDto[]> => {
  // TODO: Implement API call to fetch bookings for a specific customer
  // For now, return dummy data
  return [
    {
      id: 1,
      customerId: customerId,
      customerName: 'Test Customer',
      providerId: 1,
      providerName: 'Test Provider',
      serviceId: 10,
      serviceName: 'Test Service',
      slotStartTime: new Date().toISOString(),
      slotEndTime: new Date(Date.now() + 3600000).toISOString(),
      status: 'Confirmed',
    },
  ];
};

// Add other booking-related API calls as needed

File: legacy/WebApi/Models/Entities/Booking.cs
// WebApi/Models/Entities/Booking.cs
using System;
using System.Collections.Generic;

namespace WebApi.Models.Entities
{
    public enum BookingStatus
    {
        Pending,
        Confirmed,
        Completed,
        Cancelled
    }

    public class Booking : BaseEntity
    {
        public int CustomerId { get; set; }
        public virtual User Customer { get; set; }

        public int ProviderId { get; set; }
        public virtual User Provider { get; set; }

        public int ServiceId { get; set; }
        public virtual Service Service { get; set; }

        public int AvailabilitySlotId { get; set; }
        public virtual AvailabilitySlot AvailabilitySlot { get; set; }

        public BookingStatus Status { get; set; }

        public string ExtendedProperties { get; set; }
    }
}
